src/pivot-table/components/Wrapper.tsx

```diff
@@ -1,30 +1,47 @@
 import type { stardust } from "@nebula.js/stardust";
+import { PaginationFooter } from "@qlik/nebula-table-utils/lib/components";
 import React from "react";
-import type { PageInfo } from "../../types/types";
+import type { PageInfo, Rect } from "../../types/types";
+import { useBaseContext } from "../contexts/BaseProvider";
 import { useStyleContext } from "../contexts/StyleProvider";
 import Disclaimer from "./Disclaimer";
-import Pagination from "./Pagination";
 import { StickyPivotTable, type PivotTableProps } from "./PivotTable";
 
 export interface WrapperProps extends PivotTableProps {
   translator: stardust.Translator;
   updatePageInfo: (args: Partial<PageInfo>) => void;
+  rect: Rect;
 }
 
 export const Wrapper = (props: WrapperProps) => {
   const {
-    layoutService: { hasLimitedData },
+    layoutService: { hasLimitedData, layout },
     translator,
     pageInfo,
     updatePageInfo,
+    rect,
   } = props;
   const styleService = useStyleContext();
+  const { keyboard, theme, interactions } = useBaseContext();
 
   return (
     <>
       <StickyPivotTable {...props} />
       {hasLimitedData && <Disclaimer styleService={styleService} translator={translator} />}
-      {pageInfo.shouldShowPagination && <Pagination pageInfo={pageInfo} updatePageInfo={updatePageInfo} />}
+      <PaginationFooter
+        paginationNeeded={pageInfo.shouldShowPagination}
+        handleChangePage={(page) => updatePageInfo({ page })}
+        pageInfo={pageInfo}
+        totalRowCount={pageInfo.totalRowCount}
+        totalColumnCount={0}
+        totalPages={pageInfo.totalPages}
+        keyboard={keyboard}
+        translator={translator}
+        theme={theme}
+        interactions={interactions}
+        rect={rect as stardust.Rect}
+        layout={layout as unknown as EngineAPI.IGenericHyperCubeLayout}
```

Comment:
I think you don't need to convert it to unknown first since it's happy with only `as EngineAPI.IGenericHyperCubeLayout`. 

I got no type complain on it

Reply:
Will take a look at it